esrally/chart_generator.py

```diff
@@ -51,11 +53,18 @@ def index_label(race_config):
     return label
 
 
-def format_title(environment, track_name, suffix=None):
+def format_title(chart_type, environment, track_name, suffix=None, flavor=None):
+    title = "{}-{}".format(environment, track_name)
+
+    # as we generate two sets of Time Series/Nightly charts per flavor, the titles needs to be different.
+    if chart_type == TimeSeriesCharts:
```

Comment:
This smells as if it is now in the wrong place if we need to do a type test. IMHO we should implement this now as a method on the respective chart classes.

Reply:
The problem originally stemed from calling the static method `index` from within the floating function `generate_index_ops`:

https://github.com/elastic/rally/blob/c331b00faae9f544c7323ee2adeff5ea2f6390d8/esrally/chart_generator.py#L1341

I've refactored things in 5a3bedabf588681fce7aa0bb8d18789396447df9